src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,186 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
+    class Node<E> {
+        private E item;
+        private Node<E> next;
+        private Node<E> prev;
+
+        Node(Node<E> prevElement, E element, Node<E> nextElement) {
+            prev = prevElement;
+            item = element;
+            next = nextElement;
+        }
+    }
+
+    public MyLinkedList() {
+        this.size = 0;
+    }
+
     @Override
     public boolean add(T value) {
-        return false;
+        addToEnd(value);
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index >= 0 && index <= size) {
+            if (index == size) {
+                addToEnd(value);
+            } else {
+                addBefore(value, getNode(index));
+            }
+        } else {
+            throw new IndexOutOfBoundsException("Index " + index + " is out of bound");
+        }
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        boolean isAdded = false;
+        if (list.size() != 0) {
+            isAdded = true;
+            for (int i = 0; i < list.size(); i++) {
+                add(list.get(i));
+            }
+        }
+        return isAdded;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index >= 0 && index < size) {
```

Comment:
You can create private method, which will check value of your index and return Exception, instead of reapiting your code multiple times.